### ISSUE TITLE
File Completion Seems Broken

### ISSUE THREAD

**Issue by amadeus** [prabirshrestha/asyncomplete-file.vim#5]
This video shows the broken state. It seems to only work sometimes - almost randomly, and sometimes not at all.

[example video](http://wtf.amadeus.wtf/monosnap/screencast_2019-05-25_16-38-54.mp4)

I am currently running MacVim 8.1, patches 1-1270

My current asynccomplete settings are:

```viml
Plug 'prabirshrestha/async.vim'
Plug 'prabirshrestha/vim-lsp'
Plug 'prabirshrestha/asyncomplete.vim'
Plug 'prabirshrestha/asyncomplete-lsp.vim'
Plug 'prabirshrestha/asyncomplete-buffer.vim'
Plug 'prabirshrestha/asyncomplete-file.vim'

let g:lsp_highlight_references_enabled = 0
let g:lsp_highlights_enabled = 0
let g:lsp_textprop_enabled = 0
let g:lsp_diagnostics_echo_cursor = 0
let g:lsp_signs_enabled = 0
let g:lsp_virtual_text_enabled = 0

function! s:get_flowbin(server_info)
    let l:nodemodules_dir = lsp#utils#find_nearest_parent_directory(lsp#utils#get_buffer_path(), 'node_modules')
    if !empty(nodemodules_dir)
        " you might also want to verify if flow binary actually exists
        return [&shell, &shellcmdflag, l:nodemodules_dir . '/.bin/flow lsp']
    endif
    " instead of returning empty you could also return ['flow', 'lsp] to tell it to use global flow.
    return []
endfunction

au User lsp_setup call lsp#register_server({
    \ 'name': 'flow',
    \ 'cmd': function('s:get_flowbin'),
    \ 'root_uri':{server_info->lsp#utils#path_to_uri(lsp#utils#find_nearest_parent_file_directory(lsp#utils#get_buffer_path(), '.flowconfig'))},
    \ 'whitelist': ['javascript', 'javascript.jsx'],
    \ })

au User asyncomplete_setup call asyncomplete#register_source(asyncomplete#sources#file#get_source_options({
    \ 'name': 'file',
    \ 'whitelist': ['*'],
    \ 'priority': 10,
    \ 'completor': function('asyncomplete#sources#file#completor')
    \ }))
```

**Comment by amadeus** [comment]
Any thoughts on this? I've been pretty happy using asyncomplete, but this is one major area that doesn't quite work right.

**Comment by amadeus** [comment]
Did a bit of digging - and from what I can tell - it appears that whenever I type `/` character - the `asyncomplete#sources#file#completor` never gets called. It does get called when i type a `.` character. I also noticed:

```
function! asyncomplete#sources#file#get_source_options(opts)
  return extend(extend({}, a:opts), {
        \ 'refresh_pattern': '/$',
        \ })
endfunction
```

Which defines a refresh pattern as the `/` character, however even if I change the pattern to be a `.` it has no effect. Also digging around the asyncomplete source, it appears to never actually use this.

I am guessing this is because asyncomplete v2 was released, but this plugin never got updated to work with v2.

I think the fix here is that the `/` refresh pattern needs to be fixed to actually call the completion function, and it should be working again?
